tsdb/index/tsi1/measurement_block.go

```diff
@@ -321,16 +326,17 @@ func (e *MeasurementBlockElem) TagBlockOffset() int64 { return e.tagBlock.offset
 // TagBlockSize returns the size of the measurement's tag block.
 func (e *MeasurementBlockElem) TagBlockSize() int64 { return e.tagBlock.size }
 
-// SeriesID returns series ID at an index.
-func (e *MeasurementBlockElem) SeriesID(i int) uint64 {
-	return binary.BigEndian.Uint64(e.series.data[i*SeriesIDSize:])
-}
-
 // SeriesIDs returns a list of decoded series ids.
 func (e *MeasurementBlockElem) SeriesIDs() []uint64 {
-	a := make([]uint64, e.series.n)
-	for i := 0; i < int(e.series.n); i++ {
-		a[i] = e.SeriesID(i)
+	a := make([]uint64, 0, e.series.n)
+	var prev uint64
+	for data := e.series.data; len(data) > 0; {
+		delta, n := binary.Uvarint(data)
```

Comment:
For the measurement block, the average delta should be the average length of the series key.